### PR TITLE
Changing force density reset in lb

### LINKED ISSUE #2875: LB CPU+GPU: ext_force_density ignored in 1st time step

### PR DESCRIPTION
**Pull request by itischler**
Fixes #2875

Description of changes:
 - reinit params will apply new force density
 - changed ext force test case

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: testsuite/python/lb.py [hunk 1]
```diff
@@ -321,11 +321,15 @@ def test_a_ext_force_density(self):
         self.system.integrator.run(n_time_steps)
         # ext_force_density is a force density, therefore v = ext_force_density / dens * tau * (n_time_steps - 0.5)
         # (force is applied only to the second half of the first integration step)
+        # velocity includes half of the forces applied in the prev.
+        # integration step
         fluid_velocity = np.array(ext_force_density) * self.system.time_step * (
-            n_time_steps - 0.5) / self.params['dens']
-        for n in self.lbf.nodes():
+            n_time_steps + 0.5) / self.params['dens']
+        for n in list(itertools.combinations(range(int(self.system.box_l[0] / self.params['agrid'])), 3)):
```
Comment (KaiSzuttor) [inline testsuite/python/lb.py]: why don't you use the ```lbf.nodes()```?

**Comment by KaiSzuttor** [comment]
I think you have to adapt the hard-coded values for the sample tests.

**Comment by RudolfWeeber** [comment]
Lgtm. Can you please merge the Python brnach

**Comment by KaiSzuttor** [comment]
i've included your PR in #2992

**Pull request closed without merging**